src/structures/sgx.rs

```diff
@@ -0,0 +1,63 @@
+//! Provides architectural structures for Intel SGX.
+/// Page type
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u8)]
+#[non_exhaustive]
+pub enum PageClass {
+    /// SGX Enclave Control Structure (SECS)
+    Secs = 0,
+    /// Thread Control Structure (TCS)
+    Tcs = 1,
+    /// Regular page
+    Reg = 2,
+    /// Version Array (VA) page
+    Va = 3,
+    /// Trimmed page (aka can be removed)
+    Trim = 4,
+}
+
+bitflags::bitflags! {
+    /// Page protection bits
+    #[repr(transparent)]
+    #[non_exhaustive]
```

Comment:
`#[non_exhaustive]` doesn't do anything on types generated by `bitflags!`.

Reply:
OK, I'll remove it.